sql/sql_optimizer.cc

```diff
@@ -10144,15 +10146,17 @@ static bool internal_remove_eq_conds(THD *thd, Item *cond,
   if (cond->type() == Item::COND_ITEM)
   {
     Item_cond *const item_cond= down_cast<Item_cond *>(cond);
-    const bool and_level= item_cond->functype() == Item_func::COND_AND_FUNC;
+    const auto& item_func_type = item_cond->functype();
```

Comment:
I think that  we just port the code from mysql 5.7 is a bettern solution. The following code is not deal with all the possible conditions. 

MySQL 5.7 code for you:
https://github.com/RingsC/mysql-server/blob/5.7/sql/sql_optimizer.cc

Reply:
I think we'd better restore the sql_optimizer.cc file to the official one, the code logic being modified is too weird

Reply:
However, before the thorough separation of query optimization and tianmu logic, further separation is difficult to achieve